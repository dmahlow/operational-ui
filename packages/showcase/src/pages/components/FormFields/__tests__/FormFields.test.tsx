import * as React from "react"
import { render } from "enzyme"

import FormFieldsPage from "../FormFields"

describe("FormFieldsPage", () => {
  it("Should render correctly", () => {
    expect(render(<FormFieldsPage />)).toMatchSnapshot()
  })
})
