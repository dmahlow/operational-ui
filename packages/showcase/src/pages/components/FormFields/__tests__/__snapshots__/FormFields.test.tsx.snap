// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldsPage Should render correctly 1`] = `
<div>
  <div
    class="css-by30gj"
  >
    Form Fields
  </div>
  <p>
    Our applications tend to be composed of a myriad of form controls.
    <br />
    This page seeks to document the vast majority of these controls.
  </p>
  <div
    class="css-by30gj"
  >
    Input
  </div>
  <p>
    This component lives up to its name, allowing a user to input text, numbers, or other data.
  </p>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          placeholder
        </td>
        <td>
          Text displayed when the input field has no value.
        </td>
        <td>
          ""
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          name
        </td>
        <td>
          The name used to refer to the input, for forms.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="css-by30gj"
  >
    Select
  </div>
  <p>
    The Select component presents users with a list of information with single-choice or multiple-choice options. Select elements can have options filled onClick, and also support filters.
  </p>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Return Value
  </h4>
  <p>
    The Select component holds its value in its \`state\`. The value is either an \`Option\` object, or an Array of \`Option\` objects, depending on the \`multiple\` prop. The shape of an Option object is described below.
  </p>
  <pre
    style="display:block;overflow-x:auto;padding:0.5em;background:#F0F0F0;color:#444;"
  >
    <code>
      
Option {
    
      <span>
        id
      </span>
      : 
      <span
        style="color:#880000;"
      >
        1
      </span>
      , 
      <span
        style="color:#888888;"
      >
        // a number that is passed as a \`key\` in the React iterator,
      </span>
      
    label: 
      <span
        style="color:#880000;"
      >
        "Any string"
      </span>
      ,
    
      <span>
        value
      </span>
      : (
      <span>
        <span>
          &lt;
          <span
            style="font-weight:bold;"
          >
            div
          </span>
          &gt;
        </span>
        Literally anything can go here
        <span>
          &lt;/
          <span
            style="font-weight:bold;"
          >
            div
          </span>
          &gt;
        </span>
      </span>
      )
}

    </code>
  </pre>
  <h4>
    Props
  </h4>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          options
        </td>
        <td>
          An array of options to present to the user. This can be an empty array that is later populated onClick of the Select component using the onClick prop (described below) to fetch options beforehand.
        </td>
        <td>
          []
        </td>
        <td>
          Array
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          A function that is called before the Select component opens. Useful for retrieving options to present to the user.
        </td>
        <td />
        <td>
          func
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          placeholder
        </td>
        <td>
          A string displayed to the user when nothing is selected
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          disabled
        </td>
        <td>
          Is the select box disabled?
        </td>
        <td>
          false
        </td>
        <td>
          boolean
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          multiple
        </td>
        <td>
          Is it possible to select more than one option? This turns the \`value\` of the Select component into an array of Options instead of a single Option object
        </td>
        <td>
          false
        </td>
        <td>
          boolean
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          filterable
        </td>
        <td>
          Can the options be filtered?
        </td>
        <td>
          false
        </td>
        <td>
          boolean
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          onFilter
        </td>
        <td>
          A function that is invoked before the options are filtered. Useful to fetch new options based on the filter.
        </td>
        <td />
        <td>
          func
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
